lib/libzfs/libzfs_util.c

```diff
@@ -849,8 +849,10 @@ libzfs_read_stdout_from_fd(int fd, char **lines[])
 	char **tmp_lines = NULL, **tmp;
 
 	fp = fdopen(fd, "r");
-	if (fp == NULL)
+	if (fp == NULL) {
+		close(fd);
 		return (0);
+	}
```

Comment:
It seems like it would make more sense to close this in `libzfs_run_process_impl()` after the `libzfs_read_stdout_from_fd() call`.  Why did you instead opt to close it here?

Reply:
It's closed at the end via `fclose(fp)` if this allocation succeeds, and, crucially, once you fdopen, you cannot consume the FILE* without the fd, so it *must* be closed by this function.

Reply:
Ahh, I see.  Thanks.